File: python/숫자의표현.py
def solution(n):
    answer = 0
    
    sum, i, index = 0, 1, 2
    while i < n:
        sum += i
        if sum >= n:
            i = index
            index += 1
            if sum == n:
                answer += 1
            sum = 0
        i += 1
    
    return answer+1

File: JavaScript/숫자의표현.js
function solution(n) {
    let answer = 0;
    for (let i = 1; i <= n; i++) {
        let sum = 0, add = i, preventError = 0;
        while (sum < n && preventError < 100) {
            sum += add;
            add += 1;
            preventError += 1;
        }
        if (sum === n) answer += 1;
    }
    return answer;
}
